docs/core-concepts/run.mdx

```diff
@@ -7,48 +7,107 @@ toc_min_heading_level: 2
 
-You can specify the UDF to run in the first `udf` parameter of `fused.run` by name, tokens, or a UDF object.
+```python showLineNumbers
+import fused
+fused.run("UDF_Single_point_Eiffel_Tower")
+```
+
+import ImgSimpleETUDF from '@site/static/img/core-concepts/run-udfs/running_simple_eiffel_tower_udf.png';
+
+<div style={{textAlign: 'center'}}>
+<img src={ImgSimpleETUDF} alt="Simple UDF fused.run() returning a geodataframe" style={{width: 400}} />
+</div>
+
+There are a few other ways to run a UDF:
+- [By name from your account](/core-concepts/run/#name-from-your-account)
+- [By public UDF name](/core-concepts/run/#public-udf-name)
+- [Using a token](/core-concepts/run/#token)
+- [Using a `udf` object](/core-concepts/run/#udf-object)
+- [From Github URL](/core-concepts/run/#github-url)
+- [From git commit hash (most recommended for teams)](/core-concepts/run/#git-commit-hash-recommended-for-teams)
+
+### _Name_ (from your account)
+
+_When to use: When calling a UDF you made, from your own account._
+
+You can call any UDFs you have made simply by calling them by the name you have given your UDF when saving it
+
+(Note: This requires authentication)
 
-#### _Name_
+import ImgHelloWorldUDF from '@site/static/img/core-concepts/run-udfs/hello_world_bbox_udf.png';
 
-**UDF name.** Use this to run UDFs in your account. This requires authentication.
+<div style={{textAlign: 'center'}}>
+<img src={ImgHelloWorldUDF} alt="Hello World UDF" style={{width: 700}} />
+</div>
+
+This UDF can then be run in, say a notebook locally granted that you have authenticated:
 
 ```python showLineNumbers
-fused.run("My_Udf")
+fused.run("Hello_World_bbox")
 ```
 
-#### _Tokens_
+import ImgRunHWUDF from '@site/static/img/core-concepts/run-udfs/running_hello_world_bbox_udf.png';
+
+<div style={{textAlign: 'center'}}>
+<img src={ImgRunHWUDF} alt="RunningHello World UDF" style={{width: 700}} />
+</div>
+
+### Public UDF Name
+
+_When to run: Whenever you want to run a public UDF for free from anywhere_
 
-**Public token.** UDFs in the [public repo](https://github.com/fusedio/udfs/tree/main) may be run for free and without authentication. Reference them by prefixing their name with `UDF_`. For example, the UDF `Get_Isochrone` is run with `UDF_Get_Isochrone`.
+Any UDF saved in the [public UDF repo](https://github.com/fusedio/udfs/tree/main) can be run for free
+
+Reference them by prefixing their name with `UDF_`. For example, [the public UDF `Get_Isochrone`](https://github.com/fusedio/udfs/tree/main/public/Get_Isochrone) is run with `UDF_Get_Isochrone`:
 
 ```python showLineNumbers
 fused.run('UDF_Get_Isochrone')
 ```
 
+### Token
+
+_When to use: Whenever you want someone to be able to execute a UDF but might not want to share the name of the UDF with them_
```

Comment:
not want to share the name _or code_ ?

Reply:
Ah yes, we also can't share the code, as per [this answer form Isaac](https://www.notion.so/fused-load-fsh_-should-have-a-more-descriptive-error-message-158899d3b7638048a540d4472015ae51?d=159899d3b763804ea0e3001c90d2595c&pvs=4)

I'll update, good catch!